### ISSUE TITLE
Crash after adding Title to Project/Timeline & Black Viewer

### ISSUE THREAD

**Issue by nick314** [CommandPost/FCPCafe#231]
Apple Feedback Assistant ID: MISSING
---

Adding a title to a project timeline cause 10.6.7 to crash. Unfortunately couldn't repeat the behaviour. Also get the black viewer issue when working on a project for a while. Usually once every 15-20 minutes.

All titles used are native, no 3rd party. All libraries and footage located either internally, or external Samsung T7 SSD.
[Archive.zip](https://github.com/CommandPost/FCPCafe/files/12180646/Archive.zip)

Mac os 13.4.1 (c) 
Safari
Macbook M2 Max 16" 64 GB 2 TB

**Comment by latenitefilms** [comment]
Interestingly, two of these are the same crashes as reported in #223 - `PGPerThreadSetCurrentContextSentry` in the `ProGraphics` Framework.

```lua
Process:               Final Cut Pro [43096]
Path:                  /Applications/Final Cut Pro.app/Contents/MacOS/Final Cut Pro
Identifier:            com.apple.FinalCut
Version:               10.6.7 (408080)
Build Info:            ProEditor-40800080016000000~4 (20A104)
App Item ID:           424389933
App External ID:       858081843
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2023-07-22 13:30:05.7864 +0300
OS Version:            macOS 13.4.1 (22F770820d)
Report Version:        12
Anonymous UUID:        4E0E1615-5B7B-F894-DE08-71F731F79337

Sleep/Wake UUID:       56D98522-A9FC-48ED-B4F9-AEAE6F16D708

Time Awake Since Boot: 240000 seconds
Time Since Wake:       5596 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Codes:       0x0000000000000001, 0x0000000000000000

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [43096]

VM Region Info: 0 is not in any region.  Bytes before following region: 4336205824
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      102754000-102830000    [  880K] r-x/r-x SM=COW  ...Final Cut Pro

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   ProGraphics                   	       0x114e32c88 PGPerThreadSetCurrentContextSentry::PGPerThreadSetCurrentContextSentry() + 24
1   Ozone                         	       0x1160c88b0 OZXFreeDocument + 72
2   MotionEffect                  	       0x1148f99f4 -[MEDocumentWrapper dealloc] + 40
3   libcache.dylib                	       0x1ae21b4dc _value_entry_remove + 120
4   libcache.dylib                	       0x1ae21b3c4 _entry_remove + 172
5   libcache.dylib                	       0x1ae21af00 _cache_remove_all_locked + 76
6   libcache.dylib                	       0x1ae21ae78 cache_remove_all + 64
7   CoreFoundation                	       0x1a29d6e34 -[NSCache dealloc] + 84
8   MotionEffect                  	       0x1148f9ba4 -[MEDocumentCacheManager dealloc] + 28
9   MotionEffect                  	       0x1148f9aec +[MEDocumentCacheManager releaseSharedInstance] + 36
10  CoreFoundation                	       0x1a2967180 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 148
11  CoreFoundation                	       0x1a2a02eec ___CFXRegistrationPost_block_invoke + 88
12  CoreFoundation                	       0x1a2a02e34 _CFXRegistrationPost + 440
13  CoreFoundation                	       0x1a29384cc _CFXNotificationPost + 704
14  Foundation                    	       0x1a38968f4 -[NSNotificationCenter postNotificationName:object:userInfo:] + 88
15  AppKit                        	       0x1a5da5488 -[NSApplication terminate:] + 1092
16  LunaKit                       	       0x102a7ebcc -[LKApplication terminate:] + 84
17  Flexo                         	       0x10998a958 -[FFApplication terminate:] + 484
18  LunaKit                       	       0x102a4255c -[LKCommandsController performCommand:withTarget:] + 204
19  LunaKit                       	       0x102a42418 -[LKCommandsController performKeyEquivalent:withTarget:] + 156
20  LunaKit                       	       0x102a7ea2c -[LKApplication _handleKeyEquivalent:] + 196
21  Flexo                         	       0x10998a764 -[FFApplication _handleKeyEquivalent:] + 84
22  AppKit                        	       0x1a5cbdb84 -[NSApplication(NSEvent) sendEvent:] + 652
23  LunaKit                       	       0x102a7e930 -[LKApplication sendEvent:] + 60
24  Flexo                         	       0x10998a5e8 -[FFApplication sendEvent:] + 340
25  AppKit                        	       0x1a5f0dfc4 -[NSApplication _handleEvent:] + 60
26  AppKit                        	       0x1a5b85368 -[NSApplication run] + 500
27  LunaKit                       	       0x102a7fbd4 LKApplicationMain + 204
28  dyld                          	       0x1a253bf28 start + 2236
```

```lua
Process:               Final Cut Pro [55439]
Path:                  /Applications/Final Cut Pro.app/Contents/MacOS/Final Cut Pro
Identifier:            com.apple.FinalCut
Version:               10.6.7 (408080)
Build Info:            ProEditor-40800080016000000~4 (20A104)
App Item ID:           424389933
App External ID:       858081843
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2023-07-22 16:06:42.1949 +0300
OS Version:            macOS 13.4.1 (22F770820d)
Report Version:        12
Anonymous UUID:        4E0E1615-5B7B-F894-DE08-71F731F79337

Sleep/Wake UUID:       0746633C-1C49-4E98-A24A-BA427DB41B2F

Time Awake Since Boot: 250000 seconds
Time Since Wake:       7674 seconds

System Integrity Protection: enabled

Crashed Thread:        60  BGTask: FFDominantMotionMediaRep  Dispatch queue: analyzeAndOptimizeTaskSerializationGroup (QOS: UNSPECIFIED)

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000018
Exception Codes:       0x0000000000000001, 0x0000000000000018

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [55439]

VM Region Info: 0x18 is not in any region.  Bytes before following region: 105553518919656
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      MALLOC_NANO              600018000000-600020000000 [128.0M] rw-/rwx SM=PRV  

Thread 0::  Dispatch queue: com.apple.main-thread
0   libsystem_platform.dylib      	       0x1a28c0314 _platform_memcmp + 100
1   CoreFoundation                	       0x1a294e008 __CFBinaryPlistGetOffsetForValueFromDictionary3 + 1576
2   Foundation                    	       0x1a389dff0 _decodeObjectBinary + 824
3   Foundation                    	       0x1a38a01dc -[NSKeyedUnarchiver _decodeArrayOfObjectsForKey:] + 1592
4   Foundation                    	       0x1a389f434 -[NSDictionary(NSDictionary) initWithCoder:] + 152
5   Foundation                    	       0x1a389e63c _decodeObjectBinary + 2436
6   Foundation                    	       0x1a389db74 _decodeObject + 256
7   Foundation                    	       0x1a389d9f4 -[NSKeyedUnarchiver decodeObjectForKey:] + 216
8   Foundation                    	       0x1a389d82c -[NSKeyedUnarchiver decodeObjectOfClasses:forKey:] + 376
9   Foundation                    	       0x1a3d9d78c -[NSCoder __tryDecodeObjectForKey:error:decodeBlock:] + 96
10  Foundation                    	       0x1a389d618 -[NSCoder decodeTopLevelObjectOfClasses:forKey:error:] + 128
11  Flexo                         	       0x109a827e4 +[NSKeyedUnarchiver(FFSecureKeyedArchiver) secureUnarchivedObjectOfClasses:fromData:error:] + 112
12  Flexo                         	       0x109597448 -[FFDominantMotionMediaRep analysisDataForRange:track360:] + 148
13  Flexo                         	       0x10959b070 -[FFDominantMotionMediaRep alreadyAnalyzedRange:checkInertiaCam:lowConfidence:track360:] + 156
14  Flexo                         	       0x1095a38ec -[FFStabilizationEffect analysisDataReady:] + 380
15  Flexo                         	       0x10959eeec -[FFStabilizationEffectNotificationHandler notification:] + 104
16  Foundation                    	       0x1a3906c64 __NSThreadPerformPerform + 264
17  CoreFoundation                	       0x1a297263c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
18  CoreFoundation                	       0x1a29725d0 __CFRunLoopDoSource0 + 176
19  CoreFoundation                	       0x1a2972340 __CFRunLoopDoSources0 + 244
20  CoreFoundation                	       0x1a2970f48 __CFRunLoopRun + 828
21  CoreFoundation                	       0x1a29704b8 CFRunLoopRunSpecific + 612
22  HIToolbox                     	       0x1ac1bac40 RunCurrentEventLoopInMode + 292
23  HIToolbox                     	       0x1ac1baa7c ReceiveNextEventCommon + 648
24  HIToolbox                     	       0x1ac1ba7d4 _BlockUntilNextEventMatchingListInModeWithFilter + 76
25  AppKit                        	       0x1a5b91d44 _DPSNextEvent + 636
26  AppKit                        	       0x1a5b90ee0 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716
27  AppKit                        	       0x1a5b85344 -[NSApplication run] + 464
28  LunaKit                       	       0x102397bd4 LKApplicationMain + 204
29  dyld                          	       0x1a253bf28 start + 2236

Thread 60 Crashed:: BGTask: FFDominantMotionMediaRep Dispatch queue: analyzeAndOptimizeTaskSerializationGroup (QOS: UNSPECIFIED)
0   Helium                        	       0x103fcf93c HGGLShaderCache::ResetMetal(HGMetalContext const*) + 24
1   Helium                        	       0x103e7e05c HGGPURenderer::~HGGPURenderer() + 280
2   Helium                        	       0x103e7e380 HGGPURenderer::~HGGPURenderer() + 12
3   HeliumSenso                   	       0x102aa0a60 soHelium::~soHelium() + 36
4   HeliumSenso                   	       0x102b84528 soWorld::Release() + 148
5   HeliumSenso                   	       0x102b9a404 HFDominantMotionTracker2::~HFDominantMotionTracker2() + 32
6   HeliumSenso                   	       0x102b9a464 HFDominantMotionTracker2::~HFDominantMotionTracker2() + 12
7   HeliumSenso                   	       0x102a9d7a4 HFDominantMotionTracker2Interface::~HFDominantMotionTracker2Interface() + 48
8   Flexo                         	       0x1097aa440 FFDestDominantMotionPrivate::Itf::~Itf() + 88
9   Flexo                         	       0x1097a83f0 -[FFDestDominantMotion dealloc] + 88
10  CoreFoundation                	       0x1a2936e44 -[__NSSingleObjectArrayI dealloc] + 64
11  libobjc.A.dylib               	       0x1a25000b4 AutoreleasePoolPage::releaseUntil(objc_object**) + 196
12  libobjc.A.dylib               	       0x1a24fcb7c objc_autoreleasePoolPop + 256
13  CoreFoundation                	       0x1a293259c _CFAutoreleasePoolPop + 32
14  Foundation                    	       0x1a38ab184 -[NSAutoreleasePool release] + 144
15  Flexo                         	       0x109599f60 -[FFDominantMotionMediaRep _runBackgroundDominantmotionAnalysisTask:onTask:] + 1912
16  Flexo                         	       0x1099a2f9c FFApplicationHandleExceptionThatShouldNotBeIgnoredInBlock + 20
17  Flexo                         	       0x1091f1730 __24-[FFBackgroundTask main]_block_invoke + 364
18  Flexo                         	       0x1099a2f9c FFApplicationHandleExceptionThatShouldNotBeIgnoredInBlock + 20
19  Flexo                         	       0x1091f1584 -[FFBackgroundTask main] + 172
20  Foundation                    	       0x1a38cd0a4 __NSOPERATION_IS_INVOKING_MAIN__ + 16
21  Foundation                    	       0x1a38cc4bc -[NSOperation start] + 708
22  Foundation                    	       0x1a38cc1f0 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 16
23  Foundation                    	       0x1a38cc0e0 __NSOQSchedule_f + 172
24  libdispatch.dylib             	       0x1a26f3518 _dispatch_block_async_invoke2 + 148
25  libdispatch.dylib             	       0x1a26e4400 _dispatch_client_callout + 20
26  libdispatch.dylib             	       0x1a26e7884 _dispatch_continuation_pop + 504
27  libdispatch.dylib             	       0x1a26e6eec _dispatch_async_redirect_invoke + 584
28  libdispatch.dylib             	       0x1a26f5e98 _dispatch_root_queue_drain + 396
29  libdispatch.dylib             	       0x1a26f66c0 _dispatch_worker_thread2 + 164
30  libsystem_pthread.dylib       	       0x1a2890038 _pthread_wqthread + 228
31  libsystem_pthread.dylib       	       0x1a288ed94 start_wqthread + 8
```

```lua
Process:               Final Cut Pro [38605]
Path:                  /Applications/Final Cut Pro.app/Contents/MacOS/Final Cut Pro
Identifier:            com.apple.FinalCut
Version:               10.6.7 (408080)
Build Info:            ProEditor-40800080016000000~4 (20A104)
App Item ID:           424389933
App External ID:       858081843
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2023-07-26 15:24:34.9751 +0300
OS Version:            macOS 13.4.1 (22F770820d)
Report Version:        12
Anonymous UUID:        4E0E1615-5B7B-F894-DE08-71F731F79337

Sleep/Wake UUID:       7C6F8160-80E6-4766-9A65-DE013B8DD557

Time Awake Since Boot: 550000 seconds
Time Since Wake:       517 seconds

System Integrity Protection: enabled

Crashed Thread:        48  Dispatch queue: com.apple.helium.rq.gpu-ru0.vs0.gcd

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Codes:       0x0000000000000001, 0x0000000000000000

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [38605]

VM Region Info: 0 is not in any region.  Bytes before following region: 4363141120
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      104104000-1041e0000    [  880K] r-x/r-x SM=COW  ...Final Cut Pro

Thread 0::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x1a2853bfc kdebug_trace + 120
1   QuartzCore                    	       0x1a9e54be8 CA::Transaction::commit() + 648
2   QuartzCore                    	       0x1aa013250 CA::Transaction::flush_as_runloop_observer(bool) + 136
3   CoreFoundation                	       0x1a29719f0 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 36
4   CoreFoundation                	       0x1a29718dc __CFRunLoopDoObservers + 532
5   CoreFoundation                	       0x1a2971010 __CFRunLoopRun + 1028
6   CoreFoundation                	       0x1a29704b8 CFRunLoopRunSpecific + 612
7   HIToolbox                     	       0x1ac1bac40 RunCurrentEventLoopInMode + 292
8   HIToolbox                     	       0x1ac1baa7c ReceiveNextEventCommon + 648
9   HIToolbox                     	       0x1ac1ba7d4 _BlockUntilNextEventMatchingListInModeWithFilter + 76
10  AppKit                        	       0x1a5b91d44 _DPSNextEvent + 636
11  AppKit                        	       0x1a5b90ee0 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716
12  AppKit                        	       0x1a5b85344 -[NSApplication run] + 464
13  LunaKit                       	       0x10442fbd4 LKApplicationMain + 204
14  dyld                          	       0x1a253bf28 start + 2236

Thread 48 Crashed::  Dispatch queue: com.apple.helium.rq.gpu-ru0.vs0.gcd
0   ProGraphics                   	       0x112152c88 PGPerThreadSetCurrentContextSentry::PGPerThreadSetCurrentContextSentry() + 24
1   TextFramework                 	       0x112c54ebc TXHeliumNode::preRenderGlyphsMetal(HGPage*) + 108
2   TextFramework                 	       0x112c57614 TXHeliumNode::RenderPageMetal(HGPage*) + 76
3   Helium                        	       0x105f22ba4 HGGPURenderer::RenderPage(HGNode*, HGPage*) + 832
4   Helium                        	       0x105f22650 HGGPURenderer::ProcessOnePage(HGNode*, HGRect, HGROIsInfo&, HGFormat, HGBuffer*, bool, bool) + 720
5   Helium                        	       0x105f21784 HGGPURenderer::ProcessNodeInternal(HGNode*, HGRect, HGFormat, HGBuffer*, bool, bool, bool) + 460
6   Helium                        	       0x105f1ef28 HGGPURenderer::GetNodeBitmap(HGNode*, HGRect, HGBuffer*, bool) + 436
7   Helium                        	       0x1060336f0 HGNode::PullMetalTextures(HGPage*, int) + 400
8   Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
9   Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
10  Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
11  Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
12  Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
13  Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
14  Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
15  Helium                        	       0x1060336a4 HGNode::PullMetalTextures(HGPage*, int) + 324
16  Helium                        	       0x106033f08 HGNode::RenderPageMetal(HGPage*) + 76
17  Helium                        	       0x105f22ba4 HGGPURenderer::RenderPage(HGNode*, HGPage*) + 832
18  Helium                        	       0x105f22650 HGGPURenderer::ProcessOnePage(HGNode*, HGRect, HGROIsInfo&, HGFormat, HGBuffer*, bool, bool) + 720
19  Helium                        	       0x105f21784 HGGPURenderer::ProcessNodeInternal(HGNode*, HGRect, HGFormat, HGBuffer*, bool, bool, bool) + 460
20  Helium                        	       0x105f1e648 HGGPURenderer::RenderTiles(HGNode*, HGRect, HGFormat, HGBitmap*, bool, bool) + 268
21  Helium                        	       0x105ff18f4 HGRenderer::RenderNode(HGNode*, HGRect, HGFormat, HGBitmap*) + 144
22  Helium                        	       0x105ff5298 HGRenderer::RenderNodes(std::__1::vector<HGRendererOutput, std::__1::allocator<HGRendererOutput>> const&, bool) + 924
23  Helium                        	       0x105ff4ac8 HGRenderer::RenderNode(HGRendererOutput const&, bool) + 100
24  Helium                        	       0x105f329b8 HGRenderExecUnit::ExecuteRenderGPU(HGRenderJob*) + 1216
25  libdispatch.dylib             	       0x1a26f3518 _dispatch_block_async_invoke2 + 148
26  libdispatch.dylib             	       0x1a26e4400 _dispatch_client_callout + 20
27  libdispatch.dylib             	       0x1a26ebb68 _dispatch_lane_serial_drain + 892
28  libdispatch.dylib             	       0x1a26ec62c _dispatch_lane_invoke + 436
29  libdispatch.dylib             	       0x1a26f7244 _dispatch_workloop_worker_thread + 648
30  libsystem_pthread.dylib       	       0x1a2890074 _pthread_wqthread + 288
31  libsystem_pthread.dylib       	       0x1a288ed94 start_wqthread + 8
```

**Comment by nick314** [comment]
Though I don’t understand these error reports like you do, that seems interesting. Hopefully Apple tracks it down soon and corrects this.

Thank you for putting in the effort for the community!


_

￼

Nikos Papadopoulos
kronik.studio <http://kronik.studio/>

**Comment by nick314** [comment]
Hi Chris,

No I’m not getting it anymore (I think it was fixed on 10.6.9). However on a side-note, I did get crashes a couple of times just by clicking the audio tab - can’t make the issue repeatable. Also I have been using FCP less lately, since we moved our workflow to DaVinci (which comes with it’s own sets of issues :) )

Always with regards,
Nikos
_

￼

Nikos Papadopoulos

E	***@***.***
M 	+30 6944 29 89 35

kronik.studio <http://kronik.studio/>
